agent/taskresource/volume/dockervolume_test.go

```diff
@@ -166,6 +221,46 @@ func TestUnmarshall(t *testing.T) {
 	assert.Equal(t, driver, unmarshalledVolume.VolumeConfig.Driver)
 	assert.Equal(t, labels, unmarshalledVolume.VolumeConfig.Labels)
 	assert.Equal(t, time.Time{}, unmarshalledVolume.GetCreatedAt())
-	assert.Equal(t, VolumeCreated, unmarshalledVolume.GetDesiredStatus())
-	assert.Equal(t, VolumeStatusNone, unmarshalledVolume.GetKnownStatus())
+	assert.Equal(t, resourcestatus.ResourceStatus(VolumeCreated), unmarshalledVolume.GetDesiredStatus())
+	assert.Equal(t, resourcestatus.ResourceStatus(VolumeStatusNone), unmarshalledVolume.GetKnownStatus())
+}
+
+func TestNewVolumeResource(t *testing.T) {
+	testCases := []struct {
+		scope         string
+		autoprovision bool
+		fail          bool
+	}{
+		{
+			"task",
```

Comment:
please add a line preceding each of these testcases with a description. you can also add a `name string` field to the struct and use that as well.